nodes/exchange/FCStd_write.py

```diff
@@ -0,0 +1,182 @@
+    from numpy import ndarray
+
+    class SvWriteFCStdNode(bpy.types.Node, SverchCustomTreeNode):
+        ''' SvWriteFCStdNode '''
+        bl_idname = 'SvWriteFCStdNode'
+        bl_label = 'Write FCStd'
+        bl_icon = 'IMPORT'
+        solid_catergory = "Inputs"
+        
+        write_update : BoolProperty(
+            name="write_update", 
+            default=True)
+
+        part_name : StringProperty(
+            name="part_name", 
+            default="part_name")
+
+        @throttled
+        def changeMode(self, context):
+
+            if self.obj_format == 'mesh':
+                if 'Verts' not in self.inputs:
+                    self.inputs.remove(self.inputs['Solid'])
+                    self.inputs.new('SvVerticesSocket', 'Verts')
+                    self.inputs.new('SvVerticesSocket', 'Faces')
+                    return
+            else:
+                if 'Solid' not in self.inputs:
+                    self.inputs.remove(self.inputs['Verts'])
+                    self.inputs.remove(self.inputs['Faces'])
+                    self.inputs.new('SvSolidSocket', 'Solid')
+                    return
+
+        
+        obj_format : EnumProperty(
+                    name='format',
+                    description='choose format',
+                    items={
+                    ('solid', 'solid', 'solid'),
+                    ('mesh', 'mesh', 'mesh')},
+                    default='solid',
+                    update=changeMode)
+
+        def draw_buttons(self, context, layout):
+
+            layout.label(text="write name:")
+            col = layout.column(align=True)
+            col.prop(self, 'part_name',text="")  
+            col.prop(self, 'obj_format',text="")     
+            col.prop(self, 'write_update')
+            if self.obj_format == 'mesh':
+                col.label(text="need triangle meshes")
+
+
+        def sv_init(self, context):
+            self.inputs.new('SvFilePathSocket', "File Path")
+
+            if self.obj_format == 'mesh':
+                self.inputs.new('SvVerticesSocket', "Verts")
+                self.inputs.new('SvStringsSocket', "Faces")
+
+            else:
+                self.inputs.new('SvSolidSocket', 'Solid')
+           
+
+        def process(self):
+
+            if not self.inputs['File Path'].is_linked:
+                return
+            
+            files = self.inputs['File Path'].sv_get()
+
+            if  not len(files[0]) == 1:
+                print ('FCStd write node support just 1 file at once')
+                return
+
+            fc_file=files[0][0]
+
+            if self.obj_format == 'mesh':
+
+                if any((self.inputs['Verts'].is_linked,self.inputs['Faces'].is_linked,self.write_update)):
+
+                    verts=self.inputs['Verts'].sv_get()
+                    pols=self.inputs['Faces'].sv_get()
+                    fc_write_parts(fc_file,verts,pols,self.part_name,None,self.obj_format)

```

Comment:
`verts_in = self.inputs['Verts'].sv_get(deepcopy=False) ` 
`pols_in = self.inputs['Faces'].sv_get(deepcopy=False)` because you don't modify data in place this makes the sv_get faster
`verts, pols= match_long_repeat([verts_in, pols_in])` to match different list lenghts (`match_long_repeat `has to be imported from data_structure)
`fc_write_parts(fc_file, verts, pols, self.part_name, None, self.obj_format)`
please try to keep PEP8 style (space after coma...)